File: src/expressionparser/IEvaluable.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package expressionparser;

import java.util.ArrayList;

/**
 *
 * @author Lomztein
 */
public interface IEvaluable extends IToken {
    
    int getPrecedence ();
    
    EvaluateResult evaluate (ArrayList<TokenData> data, int thisIndex);
    
}

File: C#/MathematicalExpressionParser/Evaluation/IEvaluable.cs
using Lomztein.MathematicalExpressionParser.Token;
using System;
using System.Collections.Generic;

namespace Lomztein.MathematicalExpressionParser.Evaluation {

    public interface IEvaluable : IToken {
    
        int Precedence { get; set; }

        (double result, int[] spendIndicies) Evaluate (List<Tuple<IToken, int>> data, int thisIndex);

    }

}
